UnitedIncome/components
# useState Hook Example With Tests

### Description
> Please provide a brief description of your changes below.

This is an exmaple of the useState hook with unit tests. I'd like to discuss implementing more of these in the component library so long as the test coverage is sufficient enough.

Right now you can run these unit tests directly using `yarn testSingle`, for whatever reason when you run the entire test suite it doesn't work correctly. This would need to be resolved before we move forward with any of these. 

### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

* The component should behave as intended.
* The unit tests should run. Once the full test suite builds the test coverage should remain the same.

### Comments
> Any additional comments that will aid with the review process.

Thanks to @labichn for insight on the workings of this. I'd love to figure out a way around the `require` usage. It seems that [Jest handles mocking hooks better](https://itnext.io/testing-components-built-using-react-hooks-with-jest-enzyme-edb87d703756), perhaps switching to using it in this library may be a good idea. Innovation sprint idea? 

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/app-*)
